utils/transformData.js

```diff
@@ -0,0 +1,87 @@
+const getDateFormat = (date) => {
+  const stringDate = `${date.getDate()}. ${
+    date.getMonth() + 1
+  }. ${date.getFullYear()}`;
+
+  return stringDate;
+};
+
+const dateOfPublicationFormat = (data) =>
+  data.map((item) => {
+    if (item.dateOfPublication) {
+      const transformedDate = getDateFormat(
+        new Date(item.dateOfPublication.toDate()),
+      );
+
+      return { ...item, dateOfPublication: transformedDate };
+    }
+
+    return item;
+  });
+
+export const transformSnapshot = (snapshot) => {
+  const transformedData = snapshot.docs.map((doc) => ({
+    ...doc.data(),
+    id: doc.id,
+  }));
+
+  return dateOfPublicationFormat(transformedData);
+};
+
+export const groupByFirstChar = (array, groupProperty) =>
+  array.reduce((acc, item) => {
+    const firstChar = item[groupProperty].charAt(0);
+
+    if (!acc[firstChar]) {
+      acc[firstChar] = []; // mutation
+    }
+
+    acc[firstChar].push(item); // even this mutates
+
+    return acc;
+  }, {});
+
+export const createSlugLink = (title) =>
+  title
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, '')
+    .replace(/\s+/g, '-')
+    .toLowerCase();
+
+export const parseKeywords = (keywordsString) => {
+  if (keywordsString) {
+    const arrayKeywords = keywordsString.split(',');
+
+    arrayKeywords.forEach((item, index) => {
+      if (item.length < 2) {
```

Comment:
Uuuh. Magic numbers! Gorgeous for newcomers to the codebase :D. Be careful, some people - when code reviewing your task f.e. - could be very sensitive.

Add some nice naming to explain why 1 is too little and 3 is too much :D. I actually don't understand

Reply:
OK, I will not explain how it works, we will pretend that you have never see it and I recreate it in a next commit.

Reply:
Looking forward